narwhal/consensus/src/bullshark.rs

```diff
@@ -280,6 +232,13 @@ impl Bullshark {
             state.last_committed_sub_dag = Some(sub_dag.clone());
 
             committed_sub_dags.push(sub_dag);
+
+            // If the leader schedule has been updated, then we'll need to recalculate any upcoming
+            // leaders for the rest of the recursive commits.
+            if self.update_leader_schedule(leader.round(), &reputation_score) {
+                // TODO: a new schedule has been produced, which means that we need to recalculate leaders
```

Comment:
Flagging this as critical for safety. If we do not check that the leaders we have planned to commit are still good we might break safety. A simpler solution is to just stop committing leaders once the schedule changes, but it will cost a bit in latency

Reply:
Absolutely agreed and thanks for pointing this out. Please find on the next PR how I've tackled this https://github.com/MystenLabs/sui/pull/12540 . I basically stop committing and re-process again the last certificate with the updated schedule information which will either keep committing with the same leaders or any updates ones. I think that was the simplest strategy so I don't have to make any special checks. Performance wise shouldn't really be any heavier